stateproof/signer.go

```diff
@@ -83,83 +80,145 @@ func (spw *Worker) nextStateProofRound(latest basics.Round) basics.Round {
 	return nextrnd
 }
 
-func (spw *Worker) signStateProof(hdr bookkeeping.BlockHeader) {
-	proto := config.Consensus[hdr.CurrentProtocol]
+func (spw *Worker) signStateProof(round basics.Round) {
+	proto, votersCommitment, err := spw.getProtoVoters(round)
```

Comment:
If state proofs are stuck in the past, and a node has just starting-up, I think it will query the database every round (starting from the last state proof) which is quite bad. Can we avoid rounds that are not in % 256 ==0 ?

Reply:
We could perhaps cache a set which contains which rounds exists in the DB, in our tracker. Is the added complexity worth it?

Reply:
decided to assume that the interval doesn't change for now, and later on, when changing the interval, solve this issue.